ml_garden/core/constants.py

```diff
@@ -0,0 +1,6 @@
+from enum import Enum
+
+
+class Task(Enum):
```

Comment:
I think this should be turned to a `str, Enum`. It makes things more interoperable and has no effect in your current implementation :). This is because enum _members_ become `str`s as well (i.e., `isinstance(Task.CLASSIFICATION, str) == True`) where they behave as their values. You can treat them as strings in many places without too much issue.

For example, users could easily pass `"classification"` and it should work, so you wouldn't need to do much conversion from setting `str` to `Enum` member. If you want to do it to validate the value, you could do this:

```python
task = Task(task)
```

1. If you get `task` as an enum member, it won't fail and it'll return itself. Remember, the enum member is a str, so `()` lookup (by value) works fine.
2. If you get `task` as a `str`, it'll fail only if the `str` is not a valid task (the `()` lookup fails). Otherwise, you get the enum member.

So you get free validation and some more flexibility.

You ca test this out by doing
```python
class A(str, Enum):
    A = "a"
     B = "b"

print(A("a"))
print(A(A.A))
print(A("other"))
```